packages/assets-controllers/src/RatesController/types.ts

```diff
@@ -0,0 +1,127 @@
+import type {
+  RestrictedControllerMessenger,
+  ControllerGetStateAction,
+  ControllerStateChangeEvent,
+} from '@metamask/base-controller';
+
+import type { fetchMultiExchangeRate as defaultFetchExchangeRate } from '../crypto-compare-service';
+import type {
+  name as ratesControllerName,
+  Cryptocurrency,
+} from './RatesController';
+
+/**
+ * Represents the conversion rates from one currency to others, including the conversion date.
+ * The `conversionRate` field is a string that maps a cryptocurrency code (e.g., "BTC") to its
+ * conversion rate. For this field we use string as the data type to avoid potential rounding
+ * errors and precision loss
+ * The `usdConversionRate` provides the conversion rate to USD as a string, or `null` if the
+ * conversion rate to USD is not available. We also use string for the same reason as stated before.
+ * The `conversionDate` is a Unix timestamp (number) indicating when the conversion rate was last updated.
+ */
+export type Rate = {
+  conversionRate: string;
+  conversionDate: number;
+  usdConversionRate?: string;
+};
+
+/**
+ * Represents the conversion rates for multiple cryptocurrencies.
+ * Each key is a string representing the cryptocurrency symbol (e.g., "BTC", "SOL"),
+ * and its value is a `Rate` object containing conversion rates from that cryptocurrency
+ * to various fiat currencies or other cryptocurrencies.
```

Comment:
Minor: but I don't think we support conversion rates from cryptocurrency to cryptocurrency here?